### ISSUE TITLE
Multiple runs of osrm-partition lead to OOM kills

### ISSUE THREAD

**Issue by oxidase** [Project-OSRM/osrm-backend#4214]
With the current master running `osrm-partition` on the already partitioned graph leads to incorrect behavior:

* for the first run [`weight_offset + 1`](https://github.com/Project-OSRM/osrm-backend/blob/master/include/partition/cell_storage.hpp#L315) equals 4052896 for bayern-latest.osm

* for the second run [`weight_offset + 1`](https://github.com/Project-OSRM/osrm-backend/blob/master/include/partition/cell_storage.hpp#L315) equals 3654768480 and `osrm-partition` is killed by OOM 

Rerunning `osrm-extract` solves the issue.

@TheMarex Could it be a regression after #4089 or it is expected and `osrm-extract` run is required before `osrm-partition`?

**Comment by TheMarex** [comment]
Seems like a regression. We probably don't renumber one of the input files of `osrm-partition` that references the NBG ids we re-write.

**Comment by miccolis** [comment]
I'm pretty sure this was resolved. @oxidase please re-open if there is anything still to do here.

**Comment by oxidase** [comment]
@miccolis it is still an issue. To reproduce
```
./osrm-extract -p ../profiles/car.lua bayern-latest.osm.pbf && \
./osrm-partition bayern-latest.osrm && \
./osrm-partition bayern-latest.osrm && \
./osrm-customize bayern-latest.osrm && \
./osrm-routed bayern-latest.osrm

...
[error] [exception] std::bad_alloc
[error] Please provide more memory or consider using a larger swapfile
```

**Comment by deniskoronchik** [comment]
@oxidase in current `master` i got:
```
./osrm-extract -p ../profiles/car.lua bayern-latest.osm.pbf
./osrm-partition bayern-latest.osrm 
./osrm-partition bayern-latest.osrm 
./osrm-customize bayern-latest.osrm
[info] Loaded edge based graph: 10356818 edges, 2795182 nodes
[info] Loading partition data took 28.0876 seconds
Segmentation fault (core dumped)
```

Compiled in Debug mode

**Comment by oxidase** [comment]
@deniskoronchik it is expected, there is an issue with second renumbering in `osrm-partition`

**Comment by oxidase** [comment]
Fixed via #4602
